python/ray/train/callbacks/logging.py

```diff
@@ -63,24 +66,10 @@ def __init__(self,
                  workers_to_log: Optional[Union[int, List[int]]] = 0) -> None:
         self._logdir = logdir
         self._filename = filename
-        self._workers_to_log = self._validate_workers_to_log(workers_to_log)
         self._log_path = None
-
-    def _validate_workers_to_log(self, workers_to_log) -> List[int]:
-        if isinstance(workers_to_log, int):
-            workers_to_log = [workers_to_log]
-
-        if workers_to_log is not None:
-            if not isinstance(workers_to_log, Iterable):
-                raise TypeError("workers_to_log must be an Iterable, got "
-                                f"{type(workers_to_log)}.")
-            if not all(isinstance(worker, int) for worker in workers_to_log):
-                raise TypeError(
-                    "All elements of workers_to_log must be integers.")
-            if len(workers_to_log) < 1:
-                raise ValueError(
-                    "At least one worker must be specified in workers_to_log.")
-        return workers_to_log
+        self._results_preprocessors = [
+            WorkersResultsPreprocessor(workers_to_log=workers_to_log)
```

Comment:
I know we've had discussions about this in the past, but looking at it again, I think that the logic for `workers_to_log` shouldn't be a part of `TrainingSingleFileLoggingCallback`. This callback should only be used to create a single logging file (as per the name) and not also be responsible for `workers_to_log`.

Reply:
To clarify, my suggestion is to move the `self._results_preprocessors = [WorkersResultsPreprocessor(workers_to_log=workers_to_log)]` line to the json callback itself, and not remove all `workers_to_log` from `TrainingSingleFileLoggingCallback`.

Reply:
I had just kept parity in this PR, and I agree 💯  that these are separate concepts that shouldn't need to be intertwined. Let me refactor this!